v03_pipeline/lib/misc/pedigree_test.py

```diff
@@ -205,6 +205,107 @@ def test_parse_lineage(self) -> None:
+                    maternal_s='sample_3',
+                    paternal_s=None,
+                    sex='F',
+                ),
+                hl.Struct(
+                    s='sample_5',
+                    maternal_s='sample_3',
+                    paternal_s=None,
+                    sex='F',
+                ),
+            ],
+        )
+        self.assertEqual(
+            Family.parse_collateral_lineage(samples),
+            {
+                'sample_1': Sample(
+                    sample_id='sample_1',
+                    sex=Sex.FEMALE,
+                    mother=None,
+                    father=None,
+                    maternal_grandmother=None,
+                    maternal_grandfather=None,
+                    paternal_grandmother=None,
+                    paternal_grandfather=None,
+                    siblings=[],
+                    half_siblings=[],
+                    aunt_nephews=[],
+                ),
+                'sample_2': Sample(
+                    sample_id='sample_2',
+                    sex=Sex.MALE,
+                    mother=None,
+                    father=None,
+                    maternal_grandmother=None,
+                    maternal_grandfather=None,
+                    paternal_grandmother=None,
+                    paternal_grandfather=None,
+                    siblings=[],
+                    half_siblings=[],
+                    aunt_nephews=[],
+                ),
+                'sample_3': Sample(
+                    sample_id='sample_3',
+                    sex=Sex.FEMALE,
+                    mother='sample_1',
+                    father='sample_2',
+                    maternal_grandmother=None,
+                    maternal_grandfather=None,
+                    paternal_grandmother=None,
+                    paternal_grandfather=None,
+                    siblings=[],
+                    half_siblings=[],
+                    aunt_nephews=[],
+                ),
+                'sample_4': Sample(
+                    sample_id='sample_4',
+                    sex=Sex.FEMALE,
+                    mother='sample_3',
+                    father=None,
+                    maternal_grandmother='sample_1',
+                    maternal_grandfather='sample_2',
+                    paternal_grandmother=None,
+                    paternal_grandfather=None,
+                    siblings=[],
+                    half_siblings=['sample_5'],
+                    aunt_nephews=[],
+                ),
+                'sample_5': Sample(
+                    sample_id='sample_5',
+                    sex=Sex.FEMALE,
+                    mother='sample_3',
+                    father=None,
+                    maternal_grandmother='sample_1',
+                    maternal_grandfather='sample_2',
+                    paternal_grandmother=None,
+                    paternal_grandfather=None,
+                    siblings=[],
+                    half_siblings=[],
```

Comment:
also half-sibling of sample_4, or is this supposed to throw an error ?

Reply:
There's a [comment here](https://github.com/broadinstitute/seqr-loading-pipelines/pull/730/files#diff-b905a71b20c6c80c544cc674af65cc044d5ff12fe2b2b2801ae10e4350bbfbf0R123-R126) about this, but it's correct as is I think.  It was a design choice to only list each relationship a single time when doing these checks to avoid having everything be present twice (i.e. parent and child, sibling and sibling).